WalletWasabi.Tests/UnitTests/BestEffortEndpointConnectorTests.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin;
+using NBitcoin.Protocol;
+using NBitcoin.Protocol.Behaviors;
+using Xunit;
+
+namespace WalletWasabi.Tests.UnitTests
+{
+	public class BestEffortEndpointConnectorTests
+	{
+		[Fact]
+		public async Task CanConnectWithDifferentModes()
+        {
+            var connector = new BestEffortEndpointConnector();
+            var nodeConnectionParameters = new NodeConnectionParameters();
+            using var nodes = new NodesGroup(Network.TestNet, nodeConnectionParameters);
+
+            async Task Connect(EndPoint endpoint)
```

Comment:
I would name this with `Async` suffix too.

Reply:
Done